File: client/china_import_app/lib/features/catalogue/domain/entities/category.dart
/// Domain representation of a product category.
class Category {
  const Category({
    required this.id,
    required this.name,
    this.imageUrl,
    this.productCount,
  });

  final String id;
  final String name;
  final String? imageUrl;
  final int? productCount;

  Category copyWith({
    String? id,
    String? name,
    String? imageUrl,
    int? productCount,
  }) {
    return Category(
      id: id ?? this.id,
      name: name ?? this.name,
      imageUrl: imageUrl ?? this.imageUrl,
      productCount: productCount ?? this.productCount,
    );
  }
}

File: server/src/ChinaImportPlatform.Api/Models/Category.cs
namespace ChinaImportPlatform.Api.Models;

/// <summary>A data-driven product category so the seller can add/rename without a release.</summary>
public class Category : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public int SortOrder { get; set; }

    public string? ImageUrl { get; set; }

    public bool IsActive { get; set; } = true;
}
